### ISSUE TITLE
How to end greedy-consume arguments?

### ISSUE THREAD

**Issue by serussell** [cosiner/flag#3]
Given:
```
type flags struct {
   Something []string `names:"-s" usage:"pass in something(s)"`
   Rest      []string `args:"true"`
}
```
how does the user end the greedy consumption of `-s` and provide non-flag arguments?  E.g., a common pattern in flag parsing libraries requires multi-value argument flags to be specified once for each argument:
```
cmd -s A -s B -s C XYZ
```
would produce `len(a.Something) == 3` and `a.Rest == []string{"XYZ"}`.  If multi-value arguments are greedy, `len(a.Something) == 4` and `a.Rest == []string{}`.  

How does a user pass in arguments to `Rest` with greedy consumption?

**Comment by cosiner** [comment]
@serussell There is no builtin support like '-' and '--' currently,  but you can just define a '-nop' flag and place it after last value of '-s', it will works. Glad to see the better way, thanks.

**Comment by cosiner** [comment]
Maybe it's not so clear, what i means is:
```Go
type flags struct {
   Something []string `names:"-s" usage:"pass in something(s)"`
   Nop      bool
   Rest      []string `args:"true"`
}
```
And then `cmd -s A -s B -s C -nop XYZ` will works.

**Comment by serussell** [comment]
One solution would be to have multi-value arguments be non-greedy.  That is:
```
cmd -s A B C
```
would not assign `A`, `B`, and `C` to `s`.  To pass multiple values to one argument, require specifying the argument multiple times.  This is the behavior of most POSIX commands (e.g., `tar`) and even the `go` command itself.  Right now, with this library, the following are equivalent:
```
cmd -s A -s B
cmd -s A B
```
but only the latter breaks the ability of the library to differentiate between multi-value and non-flag arguments.  A solution would be to simply remove the latter behavior.

**Comment by cosiner** [comment]
That looks a bit wordy if the flag name is `-something` instead of simple `-s`, actually i'd prefer long flag name instead of short so i don't need to remember what the flag means, especially for those not frequently used application.

Maybe it's better to make it an parsing option.
